### PR TITLE
feature: add nuts parameter to specify dx executable path

### PR DESCRIPTION
**Pull request by peternhale**

### CODE REVIEW

**Review by RodEsp — APPROVED** [review]
I assume if it's left blank it just uses whatever `sfdx` is available on the environment's `PATH` env var?

**Comment by peternhale** [comment]
> I assume if it's left blank it just uses whatever `sfdx` is available on the environment's `PATH` env var?

Empty should delegate the decision to testkit, I assume `bin/run` in the plugin.

**Review by amphro — APPROVED** [review]

**Review by mshanemc — APPROVED** [review]
seems right to me.  Circle always surprises me.

File: src/jobs/test-nut.yml [hunk 1]
```diff
@@ -20,6 +21,14 @@ parameters:
     description: 'By default, the latest version of the standalone CLI will be installed. To install via npm, supply a version tag such as "latest" or "6".'
     default: ''
     type: string
+  sfdx_executable_path:
+    description: "Path to sfdx executable to be used by NUTs, defaults to ''"
+    default: ''
+    type: string
+  module_version:
+    description: "The version of the module repo to checkout, '' defaults to branch/PR"
```
Comment (RodEsp) [inline src/jobs/test-nut.yml:29]: The name `module_version` and this description seem vague/confusing. Reading `The version of the module repo to checkout` makes me immediately want to go look for a repo named `module`. I'd have a hard time understanding what this was referring to without previous context. What naming this something like `repo_tag` or `repo_version`?

Comment (peternhale) [reply src/jobs/test-nut.yml:29]: @RodEsp that makes sense. I will change it to repo_tag.

**Review by RodEsp — APPROVED** [review]

**Review by amphro — APPROVED** [review]

File: src/jobs/test-nut.yml [hunk 2]
```diff
@@ -31,7 +40,20 @@ steps:
   - install-sfdx:
       version: <<parameters.sfdx_version>>
       os: <<parameters.os>>
-  - checkout
+  - when:
+      condition:
+        equal: ['', <<parameters.repo_tag>>]
```
Comment (mshanemc) [inline src/jobs/test-nut.yml:45]: what's the tag stuff used for?

**Review by mshanemc — CHANGES_REQUESTED** [review]

File: src/jobs/test-nut.yml (hunk 2 above)
Comment (peternhale) [reply src/jobs/test-nut.yml:45]: @mshanemc The repo_tag will be used to checked out the plugin source at a GH tag. The value will be the version that the plugin was in when released with sfdx-cli (or as close as possible.)

See the changes made below that adds the `git clone` when tag is not an empty string.

File: src/jobs/test-nut.yml [hunk 3]
```diff
@@ -31,7 +40,20 @@ steps:
...
+        steps:
+          - checkout
+  - when:
+      condition:
+        and:
```
[the remark this answers is not in the document]
Comment (peternhale) [reply src/jobs/test-nut.yml:50]: Will do

**Review by mshanemc — APPROVED** [review]

**Pull request closed without merging**